downsample_rs/src/lttb/scalar.rs

```diff
@@ -23,21 +23,24 @@ pub fn lttb<Tx: Num, Ty: Num>(x: ArrayView1<Tx>, y: ArrayView1<Ty>, n_out: usize
 
     for i in 0..n_out - 2 {
         // Calculate point average for next bucket (containing c).
-        let mut avg_x: Tx = Tx::default();
-        let mut avg_y: Ty = Ty::default();
+        // let mut avg_x: Tx = Tx::default();
+        // let mut avg_y: Ty = Ty::default();
+        // TODO: check the impact of using f64 (is necessary to avoid overflow)
```

Comment:
Has this been checked? @jvdd

Reply:
Not yet, but will leave this to another PR (that thoroughly profiles the code)